### PR TITLE
Issue17125 -  Header Generation Incorrectly Formats Floating Point Number

### PR DESCRIPTION
**Pull request by cristiancreteanu**
The '#' format specifier in conjunction with 'g' forces the addition of the decimal point at the end, after which the letter corresponding to the type would have been added in `floatToBuffer` of `hdrgen.d`, which resulted in incorrect headers.

### CODE REVIEW

**Comment by thewilsonator** [comment]
Please make the commit message say "Fix issue 17125: ..." so that the bot picks it up.

Also please add a test case.

**Comment by RazvanN7** [comment]
If you want to run the tests locally, you need to go to the tests directory in the root directory and type `make` - this will run all the tests, but you can select a specific category of tests, for example `make run_fail_compilation_tests` -

**Comment by thewilsonator** [comment]
The doc tester has had a network error, you'll need to give this another push.

File: src/dmd/hdrgen.d [hunk 1]
```diff
@@ -1900,7 +1900,11 @@ public:
             if (r != value) // if exact duplication
                 CTFloat.sprint(buffer.ptr, 'a', value);
         }
+
```
Comment (RazvanN7) [inline src/dmd/hdrgen.d]: This change is unrelated to the patch, please undo.

**Review by RazvanN7 — APPROVED** [review]

**Comment by RazvanN7** [comment]
The azure failures is unrelated, as this patch fixes a header generation issue.
